R/merge.R

```diff
@@ -2,37 +2,78 @@
 #############################################################################
 #' Merge two \code{GADSdat} objects into a single \code{GADSdat} object.
 #'
-#' Is a secure way to merge the data and the meta data of two \code{GADSdat} objects. Currently, only limited merging options are supported.
+#' Is a secure way to merge the data and the meta data of two \code{GADSdat} objects.
+#' Currently, only limited merging options are supported.
 #'
 #' If there are duplicate variables (except the variables specified in the \code{by} argument), these variables are removed from y.
 #' The meta data is joined for the remaining variables via \code{rbind}.
 #'
+#' The function supports automatically recoding missing values created through merging with a designated missing code
+#' (\code{missingValue}) and a value label (\code{missingValLabel}).
+#'
 #'@param x \code{GADSdat} object imported via \code{eatGADS}.
 #'@param y \code{GADSdat} object imported via \code{eatGADS}.
 #'@param by A character vector.
 #'@param all A character vector (either a full join or an inner join).
 #'@param all.x See merge.
 #'@param all.y See merge.
+#'@param missingValue A numeric value that is used to replace missing values introduced through the merge.
+#'@param missingValLabel The value label that is assigned to all variables into which \code{missingValue} is inserted.
 #'@param ... Further arguments are currently not supported but have to be included for \code{R CMD} checks.
 #'
 #'@return Returns a \code{GADSdat} object.
 #'
 #'
 #'@export
-merge.GADSdat <- function(x, y, by, all = TRUE, all.x = all, all.y = all, ...) {
+merge.GADSdat <- function(x, y, by, all = TRUE, all.x = all, all.y = all,
+                          missingValue = NULL, missingValLabel = NULL, ...) {
   check_GADSdat(x)
   check_GADSdat(y)
-  if(!is.character(by)) stop(by, " is not a character vector.")
-  if(!all(by %in% names(x$dat))) stop(by, " is not a variable in x.")
-  if(!all(by %in% names(y$dat))) stop(by, " is not a variable in y.")
-  # drop double variables from y
+  if(!is.character(by)) {
+    stop(by, " is not a character vector.")
+  }
+  check_vars_in_GADSdat(x, vars = by, argName = "by", GADSdatName = "x")
+  check_vars_in_GADSdat(y, vars = by, argName = "by", GADSdatName = "y")
+
+  # drop duplicate variables from y
   y_vars <- c(by, names(y$dat)[!names(y$dat) %in% names(x$dat)])
-  if(!length(y_vars) > length(by)) stop("y does not contain unique variables.")
+  if(!length(y_vars) > length(by)) {
+    stop("y does not contain unique variables.")

```

Comment:
Nit: I was a little bit confused by this error message as I was thinking of y names like `c("ID", "V1", "V1", "V1")`. Maybe, this would be a bit more helpful: `"y does not contain variables distinct from x."`? 🤔